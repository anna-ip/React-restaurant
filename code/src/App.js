import React, { useState } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Header from "./page-components/Header/Header";
import LandingPage from "./pages/LandingPage/LandingPage";
import MenuPage from "./pages/Menu/MenuPage";
import ReviewForm from "./pages/ReviewForm/ReviewForm";
import Footer from "./page-components/Footer/Footer";
import "./App.scss";

import menus from "./menus.json";
import ReviewsContextProvider from "./contexts/ReviewsContext";

export const MenuContext = React.createContext({ menus: [] });

const App = () => {
  const [storeMenus, setStoreMenus] = useState(menus);

  return (
    <MenuContext.Provider value={[storeMenus, setStoreMenus]}>
      <ReviewsContextProvider>
        <Router>
          <Header />
          <Switch>
            <Route exact path="/">
              <LandingPage />
            </Route>
            <Route path="/menu">
              <MenuPage />
            </Route>
            <Route path="/review">
              <ReviewForm />
            </Route>
          </Switch>
          <Footer />
        </Router>
      </ReviewsContextProvider>
    </MenuContext.Provider>
  );
};

export default App;
