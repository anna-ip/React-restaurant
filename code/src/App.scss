@import './pages/Menu/MenuPage.module.scss';
@import './pages/LandingPage/LandingPage.module.scss';
@import './pages/ReviewForm/ReviewForm.module.scss';
@import './page-components/Header/Header.module.scss';
@import './page-components/Footer/Footer.module.scss';
@import './components/Nav/Burger/Burger.module.scss';
@import './components/Nav/Menu/Menu.module.scss';
@import './components/Nav/Menu/Links/NavButtons.module.scss';
@import './components/MenuPageComponents/MenuCardItem/MenuCardItem.module.scss';
@import './components/ReviewComponents/Reviews/Reviews.module.scss';
@import './components/ReviewComponents/Card/Card.module.scss';
@import './components/ReviewComponents/Arrows/Arrows.module.scss';
@import './components//StarRating/StarRating.module.scss';
@import './ui-components/Button/Button.module.scss';
@import './ui-components/Link/Links.module.scss';
@import './styles/_variables.scss';

body {
  font-family: $font-family;
  font-weight: 500;
  background-color: $base-color;
  text-align: center;
  color: $font-color;
}
