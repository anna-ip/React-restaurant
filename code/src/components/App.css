body {
  text-align: center;
}
.header-container {
  border: 1px solid black;
}
