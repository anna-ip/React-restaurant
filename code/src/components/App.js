import React from 'react'
import './App.css'
import Header from './Header'
import LandingPage from './LandingPage/LandingPage'
import MenuPage from './MenuPage/MenuPage'
import Footer from './Footer'

// Add React router

function App() {
  return (
    <>
      <Header />
      <LandingPage />
      <MenuPage />
      <Footer />
    </>
  )
}

export default App
