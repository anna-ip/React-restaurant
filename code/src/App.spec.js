import React from "react";
import App from "./App";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

test("<App/> renders correctly using enzyme", () => {
  const tree = shallow(<App />);
  expect(toJson(tree)).toMatchSnapshot();
});
