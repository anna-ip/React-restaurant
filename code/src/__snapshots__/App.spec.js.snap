// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly using enzyme 1`] = `
<ContextProvider
  value={
    Array [
      Object {
        "menus": Array [
          Object {
            "category": "breakfast",
            "desc": "I'm baby woke mlkshk wolf bitters live-edge blue bottle, hammock freegan copper mug whatever cold-pressed ",
            "id": 1,
            "img": "pexels-ash-376464.jpg",
            "price": "250kr",
            "title": "buttermilk pancakes",
          },
          Object {
            "category": "lunch",
            "desc": "vaporware iPhone mumblecore selvage raw denim slow-carb leggings gochujang helvetica man bra'id' jianbing. Marfa thundercats ",
            "id": 2,
            "img": "pexels-jonathan-borba-2983098.jpg",
            "price": "300kr",
            "title": "diner double",
          },
          Object {
            "category": "shakes",
            "desc": "ombucha chillwave fanny pack 3 wolf moon street art photo booth before they sold out organic viral.",
            "id": 3,
            "img": "pexels-archie-binamira-1193335.jpg",
            "price": "160kr",
            "title": "godzilla milkshake",
          },
          Object {
            "category": "breakfast",
            "desc": "Shabby chic keffiyeh neutra snackwave pork belly shoreditch. Prism austin mlkshk truffaut, ",
            "id": 4,
            "img": "pexels-chan-walrus-958545.jpg",
            "price": "300kr",
            "title": "country delight",
          },
          Object {
            "category": "lunch",
            "desc": "franzen vegan pabst bicycle rights kickstarter pinterest meditation farm-to-table 90's pop-up ",
            "id": 5,
            "img": "pexels-pixabay-236813.jpg",
            "price": "150kr",
            "title": "egg attack",
          },
          Object {
            "category": "shakes",
            "desc": "Portland chicharrones ethical edison bulb, palo santo craft beer chia heirloom iPhone everyday",
            "id": 6,
            "img": "pexels-rodnae-productions-6413433.jpg",
            "price": "150kr",
            "title": "oreo dream",
          },
          Object {
            "category": "breakfast",
            "desc": "carry jianbing normcore freegan. Viral single-origin coffee live-edge, pork belly cloud bread iceland put a bird ",
            "id": 7,
            "img": "pexels-rachel-claire-5531424.jpg",
            "price": "200kr",
            "title": "bacon overflow",
          },
          Object {
            "category": "lunch",
            "desc": "on it tumblr kickstarter thundercats migas everyday carry squid palo santo leggings. Food truck truffaut",
            "id": 8,
            "img": "pexels-kevin-menajang-982612.jpg",
            "price": "60kr",
            "title": "american classic",
          },
          Object {
            "category": "shakes",
            "desc": "skateboard fam synth authentic semiotics. Live-edge lyft af, edison bulb yuccie crucifix microdosing.",
            "id": 9,
            "img": "pexels-alexander-mils-2103945.jpg",
            "price": "100kr",
            "title": "quarantine buddy",
          },
          Object {
            "category": "noodles",
            "desc": "skateboard fam synth authentic semiotics. Live-edge lyft af, edison bulb yuccie crucifix microdosing.",
            "id": 10,
            "img": "pexels-horizon-content-3727198.jpg",
            "price": "250kr",
            "title": "noodle doodle",
          },
        ],
      },
      [Function],
    ]
  }
>
  <ReviewsContextProvider>
    <BrowserRouter>
      <Header />
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <LandingPage />
        </Route>
        <Route
          path="/menu"
        >
          <MenuPage />
        </Route>
        <Route
          path="/review"
        >
          <ReviewForm />
        </Route>
      </Switch>
      <Footer />
    </BrowserRouter>
  </ReviewsContextProvider>
</ContextProvider>
`;
